### PR TITLE
Internal module for sonar-api-models

### PR DESCRIPTION
**Pull request by carlosu7**
Make a common library for Sonar API objects

### CODE REVIEW

File: core-codemods/src/test/resources/test-applications/replace-apache-defaulthttpclient/src/main/java/ai/pixee/integration/replaceapachedefaulthttpclient/ReplaceApacheDefaulthttpclientApplication.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package ai.pixee.integration.replaceapachedefaulthttpclient;
+package io.pixee.integration.replaceapachedefaulthttpclient;
```
Comment (ryandens) [inline core-codemods/src/test/resources/test-applications/replace-apache-defaulthttpclient/src/main/java/ai/pixee/integration/replaceapachedefaulthttpclient/ReplaceApacheDefaulthttpclientApplication.java:1]: is there a reason we chose to rename this package? we do own `io.pixee` but we currently don't namespace any code there - either ai.pixee or io.codemodder

File: sonar-api-model/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+plugins {
+    id("java-library")
+    id("maven-publish")
```
Comment (ryandens) [inline sonar-api-model/build.gradle.kts:3]:
```suggestion
    id("io.codemodder.java-library")
    id("io.codemodder.maven-publish")
```

we have convention plugins for these which helps standardize our usage inside this project

File: sonar-api-model/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+}
+
```
Comment (ryandens) [inline sonar-api-model/build.gradle.kts:5]:
```suggestion

description = "Library with API models for data from Sonar tools"
```

Maven Central requires a description for every module that's published

**Review by ryandens — CHANGES_REQUESTED** [review]

**Review by nahsra — APPROVED** [review]

File: sonar-api-model/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+plugins {
+    id("io.codemodder.java-library")
```
Comment (ryandens) [inline sonar-api-model/build.gradle.kts:2]: this is missing the recommended plugin 
```suggestion
    id("io.codemodder.java-library")
    id("io.codemodder.maven-publish")
```

**Review by ryandens — CHANGES_REQUESTED** [review]

**Review by ryandens — APPROVED** [review]

**Pull request merged**